orderer/common/multichannel/registrar.go

```diff
@@ -446,3 +451,67 @@ func (r *Registrar) RemoveChannel(channelID string, removeStorage bool) error {
 	//TODO
 	return errors.New("Not implemented yet")
 }
+
+// validateJoinBlock returns whether this block can be used as a join block for the channel participation API
+// as well as whether the orderer is a member of the consenter set.
+// A join block is a block of either a system channel containing a ConsortiumsConfig, or an application channel
+// containing an ApplicationConfig.
+func (r *Registrar) validateJoinBlock(channelID string, configBlock *cb.Block) (isAppChannel bool, isMember bool, err error) {
+	if !protoutil.IsConfigBlock(configBlock) {
+		return false, false, errors.New("block is not a config block")
+	}
+
+	envelope, err := protoutil.ExtractEnvelope(configBlock, 0)
+	if err != nil {
+		return false, false, err
+	}
+
+	bundle, err := channelconfig.NewBundleFromEnvelope(envelope, r.bccsp)
+	if err != nil {
+		return false, false, err
+	}
+
+	// Check channel id from join block matches
+	if bundle.ConfigtxValidator().ChannelID() != channelID {
+		return false, false, fmt.Errorf("config block channelID %s does not match passed channelID %s",
+			bundle.ConfigtxValidator().ChannelID(), channelID)
+	}
+
+	// Check channel type
+	_, isSystemChannel := bundle.ConsortiumsConfig()
+	_, isAppChannel = bundle.ApplicationConfig()
+
+	// A valid system channel has consortiums but no application group
+	// A valid application channel has an application group but no consortiums
+	if isSystemChannel == isAppChannel {
+		return false, false, errors.New("invalid channel type: must contain only one of consortiums or application group")
+	}
```

Comment:
I think this part is stricter than what fabric actually enforces. To the best of my knowledge, there is no piece of code where a system channel config is strictly not allowed to have an Application group. I will ignore it completely, but will not reject it or crash because of it. So I think something like:
```go
_, isSystemChannel := bundle.ConsortiumsConfig()
if !isSystemChannel {
   _, isAppChannel = bundle.ApplicationConfig()
  if !isAppChannel {
    return false, false, error
}
}
```
is more appropriate.

Reply:
Done